sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt

```diff
@@ -7,55 +7,69 @@ import com.mapbox.common.Cancelable
 import com.mapbox.geojson.Feature
 import com.mapbox.geojson.Geometry
 import com.mapbox.geojson.Point
+import com.mapbox.maps.exception.WorkerThreadException
 import java.util.*
 
 internal class NativeMapImpl(private val map: MapInterface) :
   MapInterface, StyleManagerInterface, ObservableInterface {
 
+  private val threadChecker = ThreadChecker()
+
   override fun setSize(size: Size) {
+    checkThread()
```

Comment:
we already have `checkNativeMap` for `MapboxMap` which is actually the public API surface - perhaps add `checkThread()` there instead? delta will be very small then

Reply:
:+1:

Reply:
added now in mapboxmap and style